test/test_crc32.cc

```diff
@@ -195,7 +201,22 @@ class crc32_variant : public ::testing::TestWithParam<crc32_test> {
     }
 };
 
+/* Specifically to test where we had dodgy alignment in the acle CRC32
+ * function. All others are either byte level access or use intrinsics
+ * that work with unaligned access */
+class crc32_align : public ::testing::TestWithParam<int> {
+public:
+    void hash(int param, crc32_func crc32) {
+        uint8_t *buf = (uint8_t*)zng_alloc(sizeof(uint8_t) * (128 + param));
+        if (buf != NULL) {
+            (void)crc32(0, buf + param, 128);
+        }
+        zng_free(buf);
+    }
+};
```

Comment:
We need to use single buffer for all offsets to ensure that all addresses are checked.

Reply:
We don't actually care what the memory contents are, we are simply testing varying alignments against the alignment sanitizer, here.  Additionally, if an allocation somehow fails we have bigger problems and it will crash, causing a failure.